lib/modules/manager/puppet/index.ts

```diff
@@ -0,0 +1,16 @@
+import { ProgrammingLanguage } from '../../../constants';
+import { GithubTagsDatasource } from '../../datasource/github-tags';
+import { PuppetForgeDatasource } from '../../datasource/puppet-forge';
+
+export { extractPackageFile } from './extract';
+
+export const language = ProgrammingLanguage.Ruby;
+
+export const defaultConfig = {
+  fileMatch: ['^Puppetfile$'],
```

Comment:
```suggestion
  fileMatch: ['(^|\/)Puppetfile$'],
```

Should we support `Puppetfile` in sub folders? Is that allowed?

Reply:
Yes, I'm guessing that this would be good to allow in subdirectories.  I haven't seen that yet in the wild, it shouldn't be very common, but allowing for custom directories especially when considering Puppet Bolt could be valuable.

Having a look at the documentation for [Bolt project directories](https://puppet.com/docs/bolt/latest/projects.html#embedded-project-directories) and following how [Bolt updates the Puppetfile](https://puppet.com/docs/bolt/latest/projects.html#how-bolt-updates-projects-to-use-module-dependency-management), I think it'd be valuable to allow this.

But implementing the `bolt-project.yaml` (which then generates a `Puppetfile`) is another PR I think.

Reply:
Puppetfile in subfolders is  now supported